libs/blocks/faas/utils.js

```diff
@@ -254,6 +254,35 @@ const beforeSubmitCallback = () => {
 };
 /* c8 ignore stop */
 
+export const afterSubmitCallback = (e) => {
+  const config = getConfig();
+  if (!e.success || !(config.faasCloseModalAfterSubmit === 'on')) return;
+  const faasForms = document.querySelectorAll('.dialog-modal .faas');
+  if (faasForms.length !== 1) return;
+  const faas = faasForms[0];
+  const dialogModal = faas.closest('.dialog-modal');
+  if (dialogModal) {
```

Comment:
I realize this was how I did it in AX 😬

If I were to write it today the first thing I would try to do differently here is to early return like in L259 and L261.